### ISSUE TITLE
R-hat for SGD

### ISSUE THREAD

**Issue by dustinvtran** [airoldilab/sgd#74]
run multiple "chains". do within sequence variance and between sequence variance to compute an R-hat for SGD. If not < 1.1, then it hasn't converged, e.g., there is something problematic with the choice of learning rate hyperparameters

**Comment by dustinvtran** [comment]
Just a note that there's active interest in this from the [Stan](https://github.com/stan-dev/stan/issues/1552) people. Andrew is interested in an "R-hat" for approximate inference techniques in general, e.g., variational inference and expectation propagation. From some discussion, there's some complications with the fact that the original technique does not immediately apply. One needs to take into account not only the variance between/within the iterations of SGD but also the variance of the SGD estimator itself. A more obvious problem is convergence at different local optima.

**Comment by dustinvtran** [comment]
Marking closed as this is more generally in #18
